### ISSUE TITLE
Infix/prefix symbols are not terms

### ISSUE THREAD

**Issue by fblanqui** [Deducteam/lambdapi#359]
with A.lp:
```
symbol A:TYPE
symbol f:A→A→A
set infix 5 "+" ≔ f
symbol g:A→A
set prefix 5 "-" := g
```
the commands `type +` and `type -` fail with `Parse error`.

**Comment by rlepigre** [comment]
This has never been part of the syntax of the tool, but I agree this might be a desirable feature. Of course, as you know, you can η-expand to get the same result.

I don't want to do extensive changes to the syntax/parser before we have sorted out the parser situation. So for now I won't do anything.

**Comment by fblanqui** [comment]
@gabrielhdt We should add in the documentation on terms that infix symbols should be applied to at least two arguments. And we should improve the current error message:
```
symbol A:TYPE;
symbol f:A → A → A;
set infix 5 "+" ≔ f;
type +
// here: [..., 5:0] Unexpected token [].
```
```

**Comment by fblanqui** [comment]
Fixed in #661.

**Comment by yannl35133** [comment]
`(+)` is not a term equivalent to the underlying function when it admits implicit arguments:
when `(+)` is parsed on its own, all implicit arguments are filled with `_` so one cannot give one's own implicit arguments afterwards.
Example
```lambdapi
constant symbol Typ : TYPE;
injective symbol η {'a: Typ} : Typ → TYPE;
notation η prefix 10.0;
constant symbol int: Typ;

symbol AccidentalOk ≔ η {int}; // η ({int})
symbol ExpectedFail ≔ η {int} int; // η ({int} int)
symbol UnexpectedFail ≔ (η) {int} int; // η {int} int
```

**Comment by fblanqui** [comment]
Right. To do this, you need to use `@+` instead. That's the difference between `(+)` and `@+`.

**Comment by yannl35133** [comment]
`@+` doesn't really do what I imagine should be available

**Comment by fblanqui** [comment]
Fixed in #674 I think.
